instagram/src/components/Authentication/LoginPage.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react'
+
+import WarningMessage from './WarningMessage';
+import {LoginContainer, LoginHeader, LoginForm, LoginInput, LoginBtn} from './LoginStyles';
+
+const CryptoJS = require('crypto-js');
+
+export default class LoginPage extends React.Component {
+
+  constructor() {
+
+    super();
+
+    this.state = {
+
+      userField: '',
+      passField: '',
+      password: '',
+      showUserMessage: false,
+      showPassMessage: false,
+      userMessage: '',
+      passMessage: '',
+      headerText: 'Log In'
+
+    }
+
+  }
+
+  logIn = e => {
+
+    e.preventDefault();
+
+    if (document.activeElement.name === 'login') {
+
+      if (!localStorage[this.state.userField]) {
+
+        this.setState({showUserMessage: true, userMessage: 'invalid username!'});
+
+      }
+
+      else if (this.state.password !== CryptoJS.AES.decrypt(localStorage[this.state.userField], 's3cr3tk3yz25').toString(CryptoJS.enc.Utf8)) {
+
+        this.setState({showPassMessage: true, passMessage: 'invalid password!'});
+
+      }
+
+      else {
+
+        this.setState({headerText: 'Logging In...'}, () => setTimeout(() => this.props.setUsername(this.state.userField), 1000));
+
+      }
+
+    }
+
+    else {
+
+      if (!localStorage[this.state.userField]) {
+
+        localStorage[this.state.userField] = CryptoJS.AES.encrypt(this.state.password, 's3cr3tk3yz25').toString();
+        this.setState({headerText: 'Signing Up...'}, () => setTimeout(() => this.props.setUsername(this.state.userField), 1000));
+
+      }
+
+      else {
+
+        this.setState({showUserMessage: true, userMessage: 'username already taken!'});
+
+      }
+
+    }
+
+  }
+
+  handleKeyPress = e => {
+
+    if (e.charCode === 13) {
+      document.querySelector('button[name="login"]').focus();
+    }
+
+  }
+
+  handleChange = e => {
+
+    if (e.target.name === 'user-field') {
+
+      if (this.state.showUserMessage)
+        this.setState({showUserMessage: false});
+
+      this.setState({userField: e.target.value});
+
+    }
+
+    else {
+
+      if (this.state.showPassMessage)
+        this.setState({showPassMessage: false});
+
+      let newPassword = '';
+
+      if (e.target.value === '')
+        newPassword = '';
+
+      else if (e.target.value[e.target.value.length - 1] !== '*' && e.target.value[e.target.value.length - 1] !== undefined)
```

Comment:
You should consider object destructuring to shorten this down, but you could also just format it differently to avoid super long lines:
```javascript
else if (
  e.target.value[e.target.value.length - 1] !== '*' && 
  e.target.value[e.target.value.length - 1] !== undefined
)
```